sfge/src/main/java/com/salesforce/graph/vertex/VertexPredicate.java

```diff
@@ -1,11 +1,22 @@
 package com.salesforce.graph.vertex;
 
+import com.salesforce.graph.symbols.SymbolProvider;
 import com.salesforce.graph.visitor.VertexPredicateVisitor;
 
 public interface VertexPredicate {
     /** Return true if the implementer is interested in this vertex. */
     boolean test(BaseSFVertex vertex);
 
+    /**
+     * Return true if the implementer is interested in this vertex.<br>
+     * <br>
+     * WARNING: the {@link SymbolProvider} parameter is experimental and may be removed in future
+     * releases. It should be used sparingly, for performance reasons.
+     */
+    default boolean test(BaseSFVertex vertex, SymbolProvider provider) {
```

Comment:
Instead of adding another method with a `default` implementation, just add the additional parameter to the existing `test` method.
Doing so will require you to make changes to the existing rules, but it has the upside of being less likely to cause errors or confusion in the future.